src/main/java/io/github/bobocodebreskul/context/registry/AnnotatedBeanDefinitionReader.java

```diff
@@ -121,10 +122,10 @@ private <T> void doRegisterBean(Class<T> beanClass, String name) {
       annotatedBeanDefinition.setPrimary(true);
     }
 
-    List<Class<?>> beanDependencies = getBeanDependencies(beanClass);
+    List<BeanDependency> dependencies = getBeanDependencies(beanClass);
```

Comment:
maybe better to do call through the Util class not do import on this

Reply:
it will look more clear what happened and how

Reply:
We use the static import from this class for other methods either, the purpose of this is a concise and readable code.
There's a debug and trace level logs in these utility method calls, which should give a clear understanding what happened and where.
Did I convince you?)

Reply:
not fully but ok